url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/Engine.java

```diff
@@ -0,0 +1,643 @@
+
+                    //no locking is ever performed during this debugging block;
+                    double outputValue = outputVariable.defuzzifyNoLocks();
+                    logger.fine(String.format("%s.output = %s",
+                            outputVariable.getName(), str(outputValue)));
+                    logger.fine(String.format("%s.fuzzy = %s",
+                            outputVariable.getName(), outputVariable.fuzzify(outputValue)));
+                    logger.fine(outputVariable.fuzzyOutput().toString());
+                    logger.fine("==========================");
+                } else {
+                    logger.fine(String.format("%s.enabled = false", outputVariable.getName()));
+                }
+            }
+        }
+        /*
+         * END: Debug information
+         */
+    }
+
+    public void configure(String conjunction, String disjunction,
+            String activation, String accumulation, String defuzzifier) {
+        configure(conjunction, disjunction, activation, accumulation, defuzzifier,
+                IntegralDefuzzifier.DEFAULT_RESOLUTION);
+    }
+
+    public void configure(String conjunction, String disjunction,
+            String activation, String accumulation, String defuzzifier, int resolution) {
+        TNormFactory tnormFactory = FactoryManager.instance().tnorm();
+        SNormFactory snormFactory = FactoryManager.instance().snorm();
+
+        TNorm objConjunction = tnormFactory.createInstance(conjunction);
+        SNorm objDisjunction = snormFactory.createInstance(disjunction);
+        TNorm objActivation = tnormFactory.createInstance(activation);
+        SNorm objAccumulation = snormFactory.createInstance(accumulation);
+
+        DefuzzifierFactory defuzzifierFactory
+                = FactoryManager.instance().defuzzifier();
+        Defuzzifier objDefuzzifier = defuzzifierFactory.createInstance(defuzzifier);
+        if (objDefuzzifier instanceof IntegralDefuzzifier) {
+            ((IntegralDefuzzifier) objDefuzzifier).setResolution(resolution);
+        }
+        configure(objConjunction, objDisjunction, objActivation, objAccumulation, objDefuzzifier);
+    }
+
+    public void configure(TNorm conjunction, SNorm disjunction,
+            TNorm activation, SNorm accumulation,
+            Defuzzifier defuzzifier) {
+        for (RuleBlock ruleblock : this.ruleBlocks) {
+            ruleblock.setConjunction(conjunction);
+            ruleblock.setDisjunction(disjunction);
+            ruleblock.setActivation(activation);
+        }
+        for (OutputVariable outputVariable : this.outputVariables) {
+            outputVariable.setDefuzzifier(defuzzifier);
+            outputVariable.fuzzyOutput().setAccumulation(accumulation);
+        }
+    }
+
+    public boolean isReady() {
+        return isReady(new StringBuilder());
+    }
+
+    public boolean isReady(StringBuilder message) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 202-211, lines 216-237 into separate methods.